### PR TITLE
Add refreshToken for HubSpot

### PR DESCRIPTION
**Pull request by dansleboby**
Add refreshToken method for Hubspot

Inspire by:
https://github.com/SocialiteProviders/Apple/blob/master/Provider.php#L289-L299

### CODE REVIEW

File: src/HubSpot/Provider.php [hunk 1]
```diff
@@ -63,4 +63,29 @@ protected function mapUserToObject(array $user)
...
+     * @see https://developers.hubspot.com/docs/api/oauth-quickstart-guide#refreshing_tokens
+     *
+     * @param string $refreshToken
+     * @return ResponseInterface
+     *
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function refreshToken($refreshToken): ResponseInterface
+    {
+        return $this->getHttpClient()->post($this->getTokenUrl(), [
```
Comment (lucasmichot) [inline src/HubSpot/Provider.php:79]:
```suggestion
        $response = $this->getHttpClient()->post($this->getTokenUrl(), [
```

File: src/HubSpot/Provider.php [hunk 2]
```diff
@@ -63,4 +63,29 @@ protected function mapUserToObject(array $user)
...
         ]);
     }
+
+    /**
+     * Acquire a new access token using the refresh token.
+     *
+     * @see https://developers.hubspot.com/docs/api/oauth-quickstart-guide#refreshing_tokens
+     *
+     * @param string $refreshToken
+     * @return ResponseInterface
```
Comment (lucasmichot) [inline src/HubSpot/Provider.php:73]:
```suggestion
     * @return array
```

File: src/HubSpot/Provider.php [hunk 3]
```diff
@@ -63,4 +63,29 @@ protected function mapUserToObject(array $user)
...
+            RequestOptions::HEADERS => [
+                'Accept' => 'application/json',
+            ],
+            RequestOptions::FORM_PARAMS => [
+                'client_id' => $this->clientId,
+                'client_secret' => $this->clientSecret,
+                'grant_type' => 'refresh_token',
+                'refresh_token' => $refreshToken
+            ]
+        ]);
```
Comment (lucasmichot) [inline src/HubSpot/Provider.php:89]:
```suggestion
        ]);
        
        return json_decode((string) $response->getBody(), true);
```

**Review by lucasmichot — CHANGES_REQUESTED** [review]
Can you return an array instead

**Comment by dansleboby** [comment]
Yes, but we should change Apple too to keep a standard

**Review by atymic — APPROVED** [review]

**Pull request merged**